carbonmark-data/lib/charts/helpers.ts

```diff
@@ -70,8 +89,43 @@ export async function prepareDailyChartData<
   }
   return chartData;
 }
+/*
+  This function takes multiple queries, resolves them to datasets (global aggregates) and merge them into data usable by recharts
+  Params:
+   - fetchFunction: The function that queries the API
+  Generics:
+   - CI: Expected type of items returned in the chart data array
+   - Q: Type of the query and mapping parameters
+*/
+export async function prepareAggregatedChartData<
+  CI extends GenericAggregatedChartDataItem,
+  Q,
+>(
+  configuration: ChartConfiguration<Q, ChartMappingParams, string | number>,
+  fetchFunction: (query: Q) => Promise<AggregatedCredits>
+): Promise<ChartData<CI>> {
+  const datasets = await Promise.all(
+    configuration.map((configurationItem) => {
+      if (configurationItem.query === undefined)
+        throw "Queries are necessary for prepareDailyChartData to fetch data";
+      return fetchFunction(configurationItem.query);
+    })
+  );
+  const chartData: ChartData<CI> = [];
+  datasets.forEach((dataset, i) => {
+    const record: CI = {} as CI;
+    const chartOptions = configuration[i].chartOptions;
+    record.id = chartOptions.id;
+    record.color = chartOptions.color;
+    record.label = chartOptions.label || chartOptions.id;
+    record.quantity = dataset.quantity;
+    chartData.push(record);
+  });
+  return chartData;
+}
 
 // Common formatters
+// TODO: This goes to lib?
```

Comment:
I'd say it should only go to lib if it's used in other modules

Reply:
Comment removed in [fd3bca0](https://github.com/KlimaDAO/klimadao/pull/1487/commits/fd3bca0fba44dae4b8fd6acd3b91bbda89540018)